nhs_wagtailadmin/static/nhs_wagtailadmin/js/revision-preview.js

```diff
@@ -0,0 +1,18 @@
+function enableRevisionPreview(formSelector, options) {
+  var $form = $(formSelector);
+  var alwaysDirty = options.alwaysDirty || false;
+  var initialData = $form.serialize();
+
+  $(".action-revision-preview").click(function(e) {
+    e.preventDefault();
+
+    var dirty = (alwaysDirty || $form.serialize() !== initialData);
+    var $this = $(this);
+
+    if (dirty) {
+      alert("Unsaved changed, please save the page before previewing it.");
+      return;
+    }
+    window.open($this.data("action"), $this.data("windowname"));
```

Comment:
Do you need to check that data attributes are there before opening window with their value?